async_service/asyncio.py

```diff
@@ -130,10 +130,15 @@ def __init__(
 
                 # block here until all service tasks have finished
                 await self._wait_service_tasks()
-            finally:
-                # signal that the service is stopping
-                self._stopping.set()
-                self.logger.debug("%s stopping", self)
+            except Exception:
+                # None of the statements above are supposed to raise exceptions (as they all rely
+                # on _run_and_manage_task() to swallow and store them in self._errors, but in case
+                # they do we probably have a bug so we log that here and move on with usual
+                # service termination.
+                self.logger.error("Unexpected exception when running service", exc_info=True)
```

Comment:
1. I think this should just be `self.logger.exception(...)` which eliminates the need to do `exc_info=True`.
2. I'm curious why this doesn't re-raise the exception since this block seems to assume something is actually wrong with the library code.

Reply:
> I think this should just be self.logger.exception(...) which eliminates the need to do exc_info=True.

Makes sense

> I'm curious why this doesn't re-raise the exception since this block seems to assume something is actually wrong with the library code.

If this code path raises an exception we won't reach the end of the method, which is where we group all collected errors in a MultiError exception

Reply:
Right, I guess my thought is that maybe we shouldn't be catching this at all then since anything that triggers this code block is a library bug that needs to be fixed and thus we shouldn't be silencing it?

Reply:
Indeed. Here we can be pretty confident this is a library error and it's probably better to propagate it even if it means other errors from code using the library could have been swallowed.